packages/ra-ui-materialui/src/input/ReferenceInput.tsx

```diff
@@ -84,6 +84,11 @@ export const ReferenceInput = (props: ReferenceInputProps) => {
         filter,
     });
 
+    if (props.validate) {
```

Comment:
We could also modify the `ReferenceInputProps` type:
```
extends Omit<InputProps, 'validate'>
```

Reply:
I don't think that would be a better DX: Developers would see a failed compilation because of `validate`, without knowing what to do instead.

Reply:
If we change the type it will be a BC, right? 
This two changes break something, so IMO we should update the types in V5 and revert this PR to put it in V5 too.

Reply:
The bug in your code wasn't caused by this change, it was revealed by this change. 

As explained, a type adjustments isn't a better DX IMO. Unless we add a "never" type and explain it with JSDoc?

Reply:
Well, yes and no. You could say there was a bug, but defining the validation prop on the `ReferenceInput` did not have any effect and it was also defined on the `<AutocompleteInput>`. The validation was working and one validation prop was unnecessary, but that's not really a bug, is it?  But now the application/page is crashing, which imho is more severe and I would call this a bug that was introduced by this change.

Reply:
You're right. We shouldn't throw an error in production-only in development. I'll open a new PR to fix it. We won't introduce a type check until v5 though, as it is a breaking change.

Reply:
Thank you :tada: